### PR TITLE
Ancillary Data Matching working

### LINKED ISSUE #5: Atmospheric Opacity
As a user, I want to  know the atmospheric dust opacity of the surface, so that I can calculate the downwelling and upwelling atmospheric contribution to surface temperature.

This concern is more applicable to THEMIS band 3 than THEMIS band 9 (although we will apply this correction to both bands for consistency), as band 9 is transparent to atmospheric dust.

### LINKED ISSUE #11: Albedo
As a user, I want to be able to provide an albedo map so that the emissivity of the surface can be calculated accurately.

Currently, the best available global albedo is from TES data at 3 km/pxl, which is significantly lower than the spatial resolution of THEMIS (100 m/pxl).  This global map is publicly available and can be provided by Robin.  TES will be the default for the algorithm.

I would like the option for the user to be able to provide a different albedo input into the algorithm at any spatial scale.

- [x] @rfergason what data do you want to use?
- [x] homogenize the data
- [x] include an option for the user to be able to provide a different albedo input and override the default.

### LINKED ISSUE #20: User Configuration
As a user, I want to be able to override the defaults so that I can get myself into trouble.

### LINKED ISSUE #23: A preprocessed ISIS3 Cube can be run through Davinci
After preprocessing the PDS data into an ISIS3 cube, it is necessary to apply the Davinci processing.  This is being done in `interpolate_krc` inside of the `processimage` func.  This func should be extracted into whatever module is being created for the ISIS3 preprocessing, the MPI boilerplate removed, and the image processed.

Inside of this func additional ISIS3 calls are being made to make decisions about what Davinci processing is necessary.  Once the Davinci parameters are determined a Davinci wrapper is called.

## Acceptance Criteria
- [x] A module exists that encapsulates the pre-processing so that a 'clean' main script can simply parse arguments and dispatch the logic off to the 'processing' module.
- [x] Given a pre-processed ISIS3 cube, the davinci processing runs on Band 3 and Band 9 resulting in 2 output cubes (or geotiffs) from Davinci with all necessary processing performed.

### LINKED ISSUE #24: Post Processing w/ ISIS3 - Ancillary Data Matching
After Davinci processing a directory should exist that contains the raw data pulled from the PDS, the pre-processed ISIS3 cube and two Davinci processed data files (band 3 and band 9).  The image files now needs to be post processed (map projected).  This is being done in `interpolate_krc`on line 155 (inside of the preprocessing).  

The previously read job configuration file (JSON) contains an ancillary data key.  For every key, if the key is a full path, the data needs to be map projected and clipped to the extent of the input data.  If the key is not a full path, the data needs to be found (search for 'montone' in `interpolate_krc` to see what I mean) and then map projected.  For this project it looks like the data that will be needed is (1) albedo, (2) emissivity, (3) dust opacity.

## Acceptance Criteria
Given a Davinci processed cube that is already map projected the following data sets are map projected and saved to the working directory (along side the original, preprocessed, davinci processed, and source map projected data):

- [x] Albedo
- [x] ~~Emissivity~~
- [x] Dust Opacity (Tau)
- [x] Slope
- [x] Slope Azimuth
- [x] Elevation

```python
            in_ref = job["ancillarydata"]

            # Resample the input reference to the requested resolution
            if "reference_resolution" in job.keys():
                out_ref = os.path.join(workingpath, os.path.splitext(in_ref)[0] + 'resampled.tif')
                xres, yres = job["reference_resolution"]
                opts = gdal.TranslateOptions(xRes=xres, yRes=yres)
                gdal.Translate(out_ref, in_ref, options=opts)
                in_ref = out_ref

            reference_dataset = io_gdal.GeoDataset(in_ref)
            reference_name = job["reference"]
            # Get the temperature set to the correct size as well as spatial resolution.
            shape = list(reference_dataset.raster_size)[::-1]
            lower, upper = reference_dataset.latlon_extent
            # Update the start and stop latitudes
            parameters['startlatitude'] = lower[0]
            parameters['stoplatitude'] = upper[0]
            clipped = os.path.join(workingpath, os.path.splitext(isiscube)[0] + '_clipped.tif')
            io_gdal.match_rasters(reference_dataset, temperature, clipped,
                                  ndv=temperature.no_data_value)
            temperature = extract_temperature(clipped)

        # Extract the ancillary data
        ancillarydata = extract_ancillary_data(job, temperature, parameters, workingpath, shape, reference_dataset)
```

### PR DESCRIPTION
**Pull request by AustinSanders**
Ancillary data is output with extents clipped to match the input
image.  All ancillary data files are configurable via the json file.
Closes #24 Closes #23 Closes #20 Closes #11 Closes #5 .

### CODE REVIEW

File: themisra/processing/processing.py [hunk 1]
```diff
@@ -182,3 +182,48 @@ def processimage(job, workingpath):
...
+            # Update the start and stop latitudes
+            parameters['startlatitude'] = lower[0]
+            parameters['stoplatitude'] = upper[0]
+            clipped = os.path.join(workingpath, os.path.splitext(isiscube)[0] + '_clipped.tif')
+            io_gdal.match_rasters(reference_dataset, temperature, clipped,
+                                  ndv=temperature.no_data_value)
+            temperature = util.extract_temperature(clipped)
+
+        # Extract the ancillary data
+        ancillarydata = util.extract_ancillary_data(job, temperature, parameters, workingpath, shape, reference_dataset)
```
Comment (acpaquette) [inline themisra/processing/processing.py]: I may not be seeing it, but should the ancillary data be saved somewhere or returned?

File: themisra/utils/utils.py [hunk 2]
```diff
@@ -9,47 +10,12 @@
 import plio.utils
 from plio.utils import log
 from plio.utils.utils import check_file_exists
+from plio.io import io_gdal, io_hdf, io_json
+from plio.date import astrodate, julian2ls, julian2season
 
 import pvl
 
-
-def process_header(job):
```
Comment (acpaquette) [inline themisra/utils/utils.py]: Im guessing this was removed due to the addition of the mapancillary method?

File: themisra/utils/utils.py [hunk 3]
```diff
@@ -233,4 +205,153 @@ def memoizer(*args, **kwargs):
...
+            del arr
+        else:
+            basename = os.path.basename(v)
+            root, extension = os.path.splitext(basename)
+            tif = os.path.join(workingpath, root + '.tif')
+
+            if v == 'montone':  # Custom dust opacity
+                startls = parameters['startlsubs'][0]
+                startmartianyear = int(parameters['startmartianyear'][0])
+                files = glob.glob('/scratch/jlaura/KRC/basemaps/tau_geotiff/MY{}*'.format(startmartianyear))
```
Comment (acpaquette) [inline themisra/utils/utils.py]: Will the hardcoded paths change or is this the only place to find the martian year files?

File: themisra/utils/utils.py (hunk 2 above)
Comment (AustinSanders) [reply themisra/utils/utils.py]: This was a duplicate function present in both utils.py and processing.py.

File: themisra/utils/utils.py (hunk 3 above)
Comment (AustinSanders) [reply themisra/utils/utils.py]: My understanding is that this is just where they live.  For anyone off the network, they'll need to be changed.

Comment (jlaura) [reply themisra/utils/utils.py]: :+1: That is correct - unfortunately, we do not make these available externally.  We could (I think) and simply do not yet.

**Comment by jlaura** [comment]
@acpaquette If this looks good after @AustinSanders changes, go ahead and merge.

File: themisra/processing/processing.py (hunk 1 above)
Comment (AustinSanders) [reply themisra/processing/processing.py]: The ancillary data is being written to file in the extract_ancillary_data function.  We probably should return the ancillary data, though.  Good catch.

**Comment by acpaquette** [comment]
Looks good, I'm going to go ahead and merge.

**Pull request merged**